src/transformers/commands/add_new_model_like.py

```diff
@@ -402,12 +403,74 @@ def get_module_from_file(module_file: Union[str, os.PathLike]) -> str:
 _re_class_func = re.compile(r"^(?:class|def)\s+([^\s:\(]+)\s*(?:\(|\:)", flags=re.MULTILINE)
 
 
+def remove_attributes(obj, attrs_to_remove=None):
+    """Remove some attributes in `obj`.
+
+    The string `obj` needs to be able to be parsed by `ast.parse`.
+    """
+    if not attrs_to_remove:
+        return obj
+
+    _parsed_obj = ast.parse(obj)
+
+    if not _parsed_obj.body:
+        return obj
+
+    # The end line number in the last previous block
+    prev_end = -1
+
+    lines = obj.split(os.linesep)
+    for parsed_obj in _parsed_obj.body:
+        # So far we only look the class definitions
+        if not isinstance(parsed_obj, ast.ClassDef):
+            continue
+
+        class_def = parsed_obj
+        attributes = class_def.body
+
+        if not attributes:
+            continue
+
+        for attribute in attributes:
+            # These are 1-based line numbers, with `end_lineno` being inclusive
+            start, end = attribute.lineno, attribute.end_lineno  # noqa: F841
```

Comment:
There is already a search on the lines, so I'd really look manually for the strings and not rely on `ast` here.

Reply:
But in general, black will ensure there are spaces between methods, so you can remove any new line above the string until you reach an empty line.

Reply:
> Why would there be a decorator on is_pipeline_test_to_skip?

I don't mean `is_pipeline_test_to_skip` itself. I mean `use string/re to remove unwanted blocks` 😃.

> you can remove any new line above the string until you reach an empty line.

Doing this, the `all_generative_model_classes` below will be removed, and even `all_model_classes` (not shown here) will be also removed, and further `class RoCBertModelTest` itself ...(if I understand your word correctly)

https://github.com/huggingface/transformers/blob/40afc762379438dabcbd44d9a252d9fe4377376d/tests/models/roc_bert/test_modeling_roc_bert.py#L574-L587

Of course, we can check `#`, check `@` and whatever check necessary to make it work - and maybe only necessary to work for these 2 attributes.

However, as mentioned in https://github.com/huggingface/transformers/pull/21999#discussion_r1132460701, I am not yet very convinced that the `ast` will cause problem in our use case (we are not using `ast` to analyze any other code, but the code of `transformers` itself. I have no extensive knowledge, but in our use case, there should be no problem. If there is issue using `ast` here, it means a python version is no longer compatible with `transformers`.

Reply:
I was giving my advise for a property, you are using it for an attribute. I feel you are not interacting in good faith here, so let's stop the debate here. I will deal with the add-new model like changes.

Reply:
> I was giving my advise for a property, you are using it for an attribute.

You are right, but this PR needs to deal with `is_pipeline_test_to_skip` and `pipeline_model_mapping`. That's why I show you the problem if we go with the suggestion. (otherwise I will have to handle `pipeline_model_mapping` in another way)

> I feel you are not interacting in good faith here

@sgugger I am not interacting in bad faith: I show you a problem we will have if we go that suggestion you mentioned when dealing `pipeline_model_mapping` that is required not to be copied.